components/public-api/typescript-common/src/public-api-converter.ts

```diff
@@ -1211,6 +1213,47 @@ export class PublicAPIConverter {
                 );
             }
         }
+
+        const capitalize = (input: string) => {
+            return input.charAt(0).toUpperCase() + input.slice(1);
+        }
+
+        // This is a hack mimicking the supervisor behavior of adding dynamic IDE tasks https://github.com/gitpod-io/gitpod/blob/e7d79c355e2cd6ac34056ea52d7bdcda45975839/components/ide-service/pkg/server/server.go#L508-L540
+        const jetbrainsIdes = prebuild.workspace.config.jetbrains;
+        if (jetbrainsIdes) {
+            for (const ide in jetbrainsIdes) {
+                const ideConfig = jetbrainsIdes[ide as keyof typeof jetbrainsIdes]!;
+                if (!ideConfig.prebuilds) {
+                    continue;
+                }
+
+                if (ideConfig.prebuilds.version === "latest" || ideConfig.prebuilds.version === "both") {
+                    tasks.push(
+                        new TaskLog({
+                            taskId: `jb-warmup-${ide}-latest`,
+                            taskLabel: `JetBrains ${capitalize(ide)} warmup (latest)`,
+                            logUrl: new URL(
+                                getPrebuildLogPath(prebuild.info.id, `${++taskIndex}`),
+                                gitpodHost,
+                            ).toString(),
+                        }),
+                    );
+                }
+                if (ideConfig.prebuilds.version === "stable" || ideConfig.prebuilds.version === "both") {
+                    tasks.push(
```

Comment:
We should aware that prebuild logs are not accessible in UI directly in workspace (just like image build logs). (users can `ls /workspace/.gitpod` anyway.

Reply:
It's true, but I think also expected from the user side. If we wanted to go a step further, we could add this to the docs.

Reply:
Added docs to https://github.com/gitpod-io/website/pull/4728 about where the user can find these logs